### ISSUE TITLE
[BUG] Supervisord keeps restarting

### ISSUE THREAD

**Issue by SebastianMuskalla** [docker-mailserver/docker-mailserver#1834]
I just migrated from `tvial/docker-mailserver`. 
Unfortunately, the server that previously worked fine does not work properly.

**If `DMS_DEBUG` is set to 0 and `SUPERVISOR_LOGLEVEL` is not set,** I get the following behavior:

1. STDOUT

```[ TASKLOG ]  Initializing setup,
[ TASKLOG ]  Checking configuration,
[ TASKLOG ]  Configuring mail server,
[ WARNING ]  Clamav is disabled. You can enable it with 'ENABLE_CLAMAV=1',
[ TASKLOG ]  Post-configuration checks,
[  FATAL  ]  Please fix your configuration. Exiting...,
```
2. No logfiles
3. Server never goes up

**If I set `DMS_DEBUG` to 1**, while keeping `SUPERVISOR_LOGLEVEL`, the **server goes up** and seems to mostly work properly.

However, I can see in `mail.log` that I have regular occurrences of the following pattern.

```Feb 24 05:30:33 mail postfix/master[1010]: terminating on signal 15
Feb 24 05:30:33 mail dovecot: master: Warning: Killed with signal 15 (by pid=7 uid=0 code=kill)
```

The PID (7 here) is the PID of the supervisor process.
After being killed, the server comes back up again and mostly works properly.

**If I set `SUPERVISOR_LOGLEVEL` to `debug`,** the STDOUT is essentially an endless sequence of

```[ TASKLOG ]  Configuring mail server,
[ TASKLOG ]  Initializing setup,
[ TASKLOG ]  Checking configuration,
[ TASKLOG ]  Configuring mail server,
[ TASKLOG ]  Initializing setup,
[ TASKLOG ]  Checking configuration,
[ TASKLOG ]  Configuring mail server,
[ TASKLOG ]  Initializing setup,
[ TASKLOG ]  Checking configuration,
[ TASKLOG ]  Configuring mail server,
[ TASKLOG ]  Initializing setup,
[ TASKLOG ]  Checking configuration,
[ TASKLOG ]  Configuring mail server,
Restarted supervisord,
[ TASKLOG ]  Initializing setup,
[ TASKLOG ]  Checking configuration,
[ TASKLOG ]  Configuring mail server,
[ TASKLOG ]  Initializing setup,
[ TASKLOG ]  Checking configuration,
[ TASKLOG ]  Configuring mail server,
[ TASKLOG ]  Initializing setup,
[ TASKLOG ]  Checking configuration,
[ TASKLOG ]  Configuring mail server,
Restarted supervisord,
```

If I set `DMS_DEBUG` to 1, I get additional output about the environment, nothing that looks even remotely  suspicious though.
In both cases, the server never goes up.

Here is the `docker-compose.yml` that the container is started with

```mail:
    image: docker.io/mailserver/docker-mailserver:latest
    hostname: mail
    domainname: <REMOVED>
    container_name: mail
    restart: unless-stopped
    dns: 1.1.1.1
    ports:
        - "25:25"   # SMTP
        - "143:143" # IMAP
        - "587:587" # SMTP StartTLS
        - "993:993" # IMAP SSL
        - "465:465" # SMPT SSL
        - "110:110" # POP3
        - "995:995" # POP3 SSL
    volumes:
        - /etc/localtime:/etc/localtime:ro # sync with host time
        - /docker/volumes/mail/mail:/var/mail
        - /docker/volumes/mail/mailstate:/var/mail-state
        - /docker/volumes/mail/config/:/tmp/docker-mailserver/
        - /docker/volumes/mail/log/:/var/log/mail
        - /etc/letsencrypt/:/etc/letsencrypt # needs to contain "folder" (symlink!) with fqdn mail.<REMOVED>
        - /docker/volumes/mail/customconfig/15-mailboxes.conf:/etc/dovecot/conf.d/15-mailboxes.conf # override mailbox config
    environment:
        - TZ=Europe/Berlin
        - ENABLE_SPAMASSASSIN=1 # spamfilter
        - SPAMASSASSIN_SPAM_TO_INBOX=1
        - MOVE_SPAM_TO_JUNK=0
        - ENABLE_CLAMAV=0 # antivirus
        - ENABLE_FAIL2BAN=1 # ban wrong login attempts
        - ONE_DIR=1
        - DMS_DEBUG=1
        # - SUPERVISOR_LOGLEVEL="debug"
        - SSL_TYPE=letsencrypt
        - ENABLE_FETCHMAIL=0 # not needed anymore
        - FETCHMAIL_POLL=120 # see above
        - REPORT_RECIPIENT=0 #
        - ENABLE_QUOTAS=0 # weird error messages cause of our custom mailbox structure
    cap_add:
        - NET_ADMIN # needed for FAIL2BAN to work
        - SYS_PTRACE
 ```

The custom mailbox structure that is mentioned above reconfigures the `namespace inbox` to move some of the default mailboxes to a subfolder.

**Comment by cybermcm** [comment]
Same problem here, slightly different: Current image, server never comes up
```
[ TASKLOG ]  Post-configuration checks,
[  FATAL  ]  Please fix your configuration. Exiting...,
```
If I set `DMS_DEBUG` to 1 server starts and works.
Difference: I do not get the error msg until now `dovecot: master: Warning: Killed with signal 15 (by pid=7 uid=0 code=kill)`

changing `SUPERVISOR_LOGLEVEL` doesn't change anything in my setup

Happy to provide more info if necessary

**Comment by aendeavor** [comment]
Yep, this is known since 2am tonight. @casperklein has already investigated and we're looking to fix this ASAP.

**Comment by aendeavor** [comment]
Just pushed `v9.0.0` again. Should be available in about an hour due to build and test time.

**Comment by cybermcm** [comment]
very quick first test, `DMS_DEBUG` and `SUPERVISOR_LOGLEVEL` not set, server is starting and receiving mails...
thx for the quick fix, I'll monitor closely today for anything unusual.

**Comment by aendeavor** [comment]
@cybermcm Alright. Thanks for the report later in advance so we know this has been fixed or not :D

**Comment by mindrunner** [comment]
Have had the same problem last night. Unfortunately, my container auto-updated and got in a non-working state without letting me know :( Reverted back to  `8.0.1` in the morning. Now trying `9.0.0` again and it seems to work! :)

Pinning version now instead of using `latest`

**Comment by cybermcm** [comment]
No problems after a few hours (with version from today in place)...
I think this can be marked as solved, although I never experienced the error described in the OP `mail dovecot: master: Warning: Killed with signal`
maybe another issue?

**Comment by casperklein** [comment]
> Pinning version now instead of using latest

@mindrunner Thats always a good practice. The latest fixed version now is 9.0.1 (9.0.0 is also fixed).

> No problems after a few hours

@cybermcm Good to hear 👍 

>  maybe another issue?

Probably. If it occurs again, please open a separate issue.
